singly_linked_list

```diff
@@ -0,0 +1,317 @@
+//ADT implementation: https://runestone.academy/ns/books/published/pythonds3/BasicDS/TheUnorderedListAbstractDataType.html
+//except for add_item as it copying append
+
+#include <stdio.h>
+#include <stdlib.h>
+
+//define MAX_SIZE; not handled but keep in mind
+
+struct Node
+{
+    int data;
+    struct Node *next;
+};
+
+static struct Node *head;
+
+
+struct Node *List(void) {
+    struct Node *p = (struct Node *) malloc(sizeof(struct Node));
+    p -> data = -1;
+    p -> next = NULL;
+    return p;
+}
+
+int is_empty() {
+    if(head -> data == -1) {
+        return 1;
+    }
+    else {
+        return 0;
+    }
+}
+
+int size() {
+    if(is_empty()) {
+        return 0;
+    }
+
+    int size = 1;
+    struct Node *node = head;
+
+    while(node -> next != NULL) {
+        size++;
+        node = node -> next;
+    }
+    return size;
+}
+
+static struct Node *add_node() {
+    return (struct Node *) malloc(sizeof(struct Node));
+}
+
+void append(int item) {
+    if(is_empty()) {
+        head -> data = item;
+    }
+    else {
+        struct Node *node = head;
+
+        while(node -> next) {
```

Comment:
Pushing to the end of the list isn't needed in case of a signly linked list

Reply:
It is a stack, clear, thanks